### PR TITLE
PORT: Makes diagonal movement euclidean

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Ports: https://github.com/Citadel-Station-13/Citadel-Station-13/pull/14911

image

To quote from the other PR:

"Right now, diagonal movement takes twice as long as orthogonal movement. In other words, in terms of mob movement, SS13 treats the tile grid as a taxicab geometry. The issue here is that much of the game doesn't. Projectiles are euclidean: it takes just as much time to move 10 meters diagonally as it does to move 10 meters orthogonally. The thing is, 10 meters diagonally is just under 7 meters north/south and 7 meters east/west, which always takes the equivalent of 14 orthogonal tile movements for units.

In other words: on diagonals, projectiles move sqrt(2) times as fast as mobs do. This fixes that."

## Why It's Good For The Game

Have two comparisons to decide:

**Without SQRT**

https://user-images.githubusercontent.com/9026500/143377073-3f86055a-75a2-4fee-a3ef-1814bfeb97d6.mp4

**With SQRT**

https://user-images.githubusercontent.com/9026500/143377110-8fc7570c-1d80-45e2-8902-9af76084b1fd.mp4

In my humble opinion, I feel like this improves movement by a large degree.

## Changelog

:cl:
balance: Diagonal movement is now slightly faster
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
do you move faster i cant tell

**Comment by Gandalf2k15** [comment]
> do you move faster i cant tell

I did a bit of in-depth comparison, what's happening is non-diagonal movement is the same, but the diagonal movement is SLIGHTLY faster.

Because I'm moving diagonally so much in the video, it LOOKS like all movement is faster, but, it's not!

**Comment by MMMiracles** [comment]
time to bring back donutstation

**Comment by necromanceranne** [comment]
> time to bring back donutstation



> time to bring back donutstation

The slow diagonal movement was not the reason donutstation sucked, my guy.

**Review by Ghommie — CHANGES_REQUESTED** [review]
It seems another line in client/Move() is using taxicab geometry:
https://github.com/tgstation/tgstation/blob/8562e9ba919b80723f60214144a8f176fda067de/code/modules/mob/mob_movement.dm#L125
Don't forget to change it to euclidean too.

**Comment by CRITAWAKETS** [comment]
being so used to the weirdly slow diagonal movement, seeing it not be slowed down just looks weird

**Comment by necromanceranne** [comment]
I think given it'll be just like moving directionally, I think it'll be fine. Maybe a bit jarring at first but nothing in this vid seems out of place, so I'm excited to see what it looks like.

I doubt we'll have a repeat of the initial rollout of smooth movement (with the buggy screen tearing causing nausea due to Byond issues).

**Comment by Twaticus** [comment]
its now been so long since diagonal movement was added that people are unaware the movement delay was intentional 
feels like yesterday

**Comment by Watermelon914** [comment]
It's not really a big balance change at all, but it does have a slight impact on the speed of diagonal movement, and I wouldn't really classify it as a code improvement either.

**Comment by TheBonded** [comment]
> It's not really a big balance change at all, but it does have a slight impact on the speed of diagonal movement, and I wouldn't really classify it as a code improvement either.

could be a refactor then? or perhaps qol?

**Comment by Gandalf2k15** [comment]
My poor gbp

**Comment by Mothblocks** [comment]
its balance

**Comment by Acer202** [comment]
This feels _really good_ please merge fully!

**Comment by LemonInTheDark** [comment]
update: turns out the reason to make it * 2 is because it means that default movespeed stays in lockstep with tick rate, avoiding weird jittering. we just didn't notice because everything jitters all the time.
